internal/provider/incident_catalog_entries_data_source.go

```diff
@@ -0,0 +1,187 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentCatalogEntriesDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentCatalogEntriesDataSource{}
+)
+
+func NewIncidentCatalogEntriesDataSource() datasource.DataSource {
+	return &IncidentCatalogEntriesDataSource{}
+}
+
+type IncidentCatalogEntriesDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentCatalogEntriesDataSourceModel struct {
+	CatalogTypeID  types.String                          `tfsdk:"catalog_type_id"`
+	CatalogEntries []IncidentCatalogEntryDataSourceModel `tfsdk:"catalog_entries"`
+}
+
+func (d *IncidentCatalogEntriesDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*IncidentProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *IncidentProviderData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client.Client
+}
+
+func (d *IncidentCatalogEntriesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_catalog_entries"
+}
+
+func (d *IncidentCatalogEntriesDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data IncidentCatalogEntriesDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var allEntries []client.CatalogEntryV3
+
+	// Fetch entries for the specified catalog type
+	catalogTypeID := data.CatalogTypeID.ValueString()
+	params := &client.CatalogV3ListEntriesParams{
+		CatalogTypeId: catalogTypeID,
+	}
+
+	// Paginate through all entries
+	params.PageSize = 100
+
+	for {
```

Comment:
without any filtering support, I wonder if this might cause some sadness if you have like 40k entries on your type 🤔 I think it's _probably_ okay though